components/director/internal/tenantmapping/handler.go

```diff
@@ -1,107 +1,143 @@
-
-		wrappedErr := errors.Wrap(err, "while decoding request body")
-		http.Error(writer, wrappedErr.Error(), http.StatusInternalServerError)
+		respondWithError(writer, http.StatusBadRequest, err, "while parsing the request")
 		return
 	}
 
-	defer func() {
-		err := request.Body.Close()
-		if err != nil {
-			wrappedErr := errors.Wrap(err, "while decoding request body")
-			log.Error(wrappedErr)
-			http.Error(writer, wrappedErr.Error(), http.StatusInternalServerError)
-		}
-	}()
+	tx, err := h.transact.Begin()
+	if err != nil {
+		respondWithError(writer, http.StatusInternalServerError, err, "while opening the db transaction")
+		return
+	}
+	defer h.transact.RollbackUnlessCommited(tx)
 
-	logBuilder.WriteString(fmt.Sprintf("\nInput: %+v", data))
+	ctx := persistence.SaveToContext(context.TODO(), tx)
 
-	if data.Extra == nil {
-		data.Extra = make(map[string]interface{})
+	credID, tenantID, refObjectType, err := h.lookupForTenant(ctx, reqData)
+	if err != nil {
+		respondWithError(writer, http.StatusInternalServerError, err, "while looking for tenant")
+		return
 	}
 
-	extraMap, ok := data.Extra.(map[string]interface{})
-	if !ok {
-		err := fmt.Errorf("Incorrect type %T; expected map[string]interface{}\n", data.Extra)
-		log.Info(logBuilder.String())
-		log.Error(err)
-		http.Error(writer, err.Error(), http.StatusInternalServerError)
+	reqData.Extra["tenant"] = tenantID
+
+	scope, err := h.lookupForScope(credID, refObjectType)
+	if err != nil {
+		respondWithError(writer, http.StatusInternalServerError, err, "while looking for scopes")
 		return
 	}
 
-	extraMap["tenant"] = "9ac609e1-7487-4aa6-b600-0904b272b11f"
+	reqData.Extra["scope"] = scope
 
-	_, ok = extraMap["scope"]
-	if !ok {
-		h.setScopes(extraMap)
+	writer.Header().Set("Content-Type", "application/json")
+	err = json.NewEncoder(writer).Encode(reqData)
+	if err != nil {
+		respondWithError(writer, http.StatusInternalServerError, err, "while encoding data")
+		return
 	}
+}
 
-	data.Extra = extraMap
+func (h *Handler) lookupForTenant(ctx context.Context, reqData ReqData) (string, string, CredenialRefObjectType, error) {
+	credID, credSrc, err := reqData.GetCredentialID()
+	if err != nil {
+		return "", "", "", errors.Wrap(err, "while determining the credential from the request")
+	}
 
-	logBuilder.WriteString(fmt.Sprintf("\nOutput: %+v\n", data))
-	log.Info(logBuilder.String())
+	switch credSrc {
+	case CertificateCredSrc, TokenCredSrc:
+		sysAuth, err := h.systemAuthSvc.GetGlobal(ctx, credID)
+		if err != nil {
+			return "", "", "", errors.Wrap(err, "while retrieving system auth from database")
+		}
 
-	err = json.NewEncoder(writer).Encode(data)
-	if err != nil {
-		wrappedErr := errors.Wrap(err, "while encoding data")
-		log.Error(wrappedErr)
-		http.Error(writer, wrappedErr.Error(), http.StatusInternalServerError)
-		return
+		refObjType := sysAuthToCredRefObjectType(*sysAuth)
+
+		if refObjType == IntegrationSystemRefObject {
+			// TODO - handle tenant for integration system
```

Comment:
Why this is not handled at all? Just get `Tenant` header and pass the value as tenant (while we don't have internal <-> external tenant mapping.